src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs

```diff
@@ -92,11 +99,32 @@ internal static class NamedPipeUtils
                 appDomainName = DefaultAppDomainName;
             }
 
-            return NamedPipeNamePrefix +
-                    proc.StartTime.ToFileTime().ToString(CultureInfo.InvariantCulture) + "." +
-                    proc.Id.ToString(CultureInfo.InvariantCulture) + "." +
-                    CleanAppDomainNameForPipeName(appDomainName) + "." +
-                    proc.ProcessName;
+            System.Text.StringBuilder pipeNameBuilder = new System.Text.StringBuilder(MaxNamedPipeNameSize);
+            pipeNameBuilder.Append(NamedPipeNamePrefix)
+#if UNIX
+                // The starttime is there to prevent another process easily guessing the pipe name
+                // and squatting on it.
+                // There is a limit of 104 characters in total including the temp path to the named pipe file
+                // on non-Windows systems, so we'll convert the starttime to hex and just take the first 8 characters.
+                .Append(proc.StartTime.ToFileTime().ToString("X8").Substring(1,8))
+#else
+                .Append(proc.StartTime.ToFileTime().ToString(CultureInfo.InvariantCulture))
```

Comment:
We want the name to be as unique as possible.  If there is no length limitation on Windows then we should use the full start time.

Reply:
As far as I know, the starttime segment of the filename is just for uniqueness and not processed.  We could have the same code for both rather than having different code for Windows vs Unix.  8 hex characters seems sufficient.  What do you think @PaulHigin ?

Reply:
As long as it is unique enough.  On Linux we limit connections to processes created by current user.  For Windows it is system wide.  I think we want to err on the side caution for Windows.

Reply:
8 hex characters gives 32-bits of uniqueness which seems sufficient.  FILETIME is int64, but that seems overkill particularly when most of the digits are likely zeros.